bin/connectors.py

```diff
@@ -1611,21 +1620,23 @@ def getPosition(self)->dict:
                             gpsposition["bearing"] = float(position["bearing"])
                             gpsposition["speed_mph"] = float(position["speed_mph"])
                             gpsposition["isvalid"] = True
+                            gpsposition["source"] = "gps"
 
                             # sanity check
                             if gpsposition["latitude"] == 0 or gpsposition["longitude"] == 0:
                                 gpsposition["isvalid"] = False
 
                 if nofix == True:
 
+
                     # we're still waiting on the GPS to obtain a fix so we wait this long
                     seconds = round((1.2) ** trycount) if trycount < 22 else (1.2) ** 22
```

Comment:
Three questions:
1) Since the outer loop that starts at 1601 tests for trycount < 2, is testing for trycount < 22 here necessary/useful?
2) What is the purpose of returning 1.2 ** 22 (i.e. 55.2061438912436) in the else?  i.e. What is the significance of 22?
3) If trycount << 22, then seconds will have data type int. If trycount >= 22 and we go through the else, the data type of seconds will be float. Will differing return types cause problems in the calling code? Perhaps the else should have a call to round()?

Reply:
That ```trycount < 22``` was a hold over from having the connectors (and subprocesses.py) do more “waiting” on the GPS to reach a 3D fix.  Turns out that’s not needed as the subprocess that we’re trying to run can deal with that (ex. Aprsc, direwolf, etc.).  For the connectors.py code, the only legitimate reason for waiting on a 3D fix (i.e. the trycount < 22 business) was for connections to APRS-IS when the brick is set to beacon to the internet.  In that case, we can’t beacon without knowing our locaton…thus the original need to monitor ```trycount``` when the number of times through that while loop became large.  

Long story short, there isn’t a need for that ```trycount < 22``` business anymore.  Good catch!!